apps/drec-api/src/utils/group-by-properties.ts

```diff
@@ -0,0 +1,48 @@
+export const arrayFromObject = <T>(obj: T): T[Extract<keyof T, string>][] => {
+  const arr = [];
+  for (const i in obj) {
+    arr.push(obj[i]);
+  }
+  return arr;
+};
+
+export const groupByProperties = <T, U, K>(
+  list: U[],
+  callback: (listItem: U) => K[],
+): T[Extract<keyof T, string>][] => {
+  const groups: any = {} as T;
+  for (let i = 0; i < list.length; i++) {
+    const group: string = JSON.stringify(callback(list[i]));
+    if (group in groups) {
+      groups[group].push(list[i]);
+    } else {
+      groups[group] = [list[i]];
+    }
+  }
+  return arrayFromObject(groups);
+};
+
+export const groupByProps = <T, U, K>(
```

Comment:
this whole function can be replaced by `values(groupBy(list, i => JSON.stringify(callback(i)))`

By the way this function looks like it's doing exactly the same thing as `groupByProperties` 🤔  in different way

Reply:
Same comment as before. And yes, the function does the same thing as groupByProperties. Initially I did the previous one and then I simplified it with this but thought it might be nice to keep both :D

Reply:
Why keep both? 🤣